src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -135,19 +135,33 @@ public PublicClientApplicationBuilder WithIosKeychainSecurityGroup(string keycha
         /// please install the nuget package Microsoft.Identity.Client.Desktop and call .WithWindowsBroker(true)</remarks>
         public PublicClientApplicationBuilder WithBroker(bool enableBroker = true)
         {
+#pragma warning disable CS0162 // Unreachable code detected
+
 #if NET45
-            throw new PlatformNotSupportedException("The Windows broker is not available on .NET Framework 4.5, please use at least .NET Framework 4.6.2");
+            throw new PlatformNotSupportedException(
+                "The Windows broker is not available on .NET Framework 4.5, please use at least .NET Framework 4.6.2");
 #endif
 
-#if NET461 || NET_CORE
+#if NET461
             if (Config.BrokerCreatorFunc == null)
             {
                 throw new PlatformNotSupportedException(
-                    "The Windows broker is not directly available on MSAL for .NET Framework or .NET Core 3.x" +
+                    "The Windows broker is not directly available on MSAL for .NET Framework" +
                     " To use it, please install the nuget package named Microsoft.Identity.Client.Desktop " +
-                    "and call the extension method .WithWindowsBroker(true)");
+                    "and call the extension method .WithDesktopFeatures() first.");
+            }
+#endif
+
+#if NET_CORE
```

Comment:
@mjcheetham - does this look right to you? When you target .net5 on Mac and Linux, it will NO-OP for now (until we have broker on Mac / Linux). On Windows, it will throw this exception.